client/app/components/parameter-settings.html

```diff
@@ -23,7 +23,7 @@ <h4 class="modal-title">{{ $ctrl.isNewParameter ? 'Add Parameter' : $ctrl.parame
       </div>
       <div class="form-group">
         <label>Type</label>
-        <select ng-model="$ctrl.parameter.type" class="form-control">
+        <select ng-model="$ctrl.parameter.type" class="form-control" ng-change="$ctrl.updateType()">
```

Comment:
We need to set isDirty for any value change, not only type. While the type change is important for validation, the rest are important for consistent UI/UX.